### PR TITLE
Paclet

### PR DESCRIPTION
**Pull request by og113**
Changed a few things, and moved a few things around, to make WallGoMatrix easily installable as a Paclet. I've done some tests locally, but the workflows and so on should be tested before merging.

### CODE REVIEW

File: .github/workflows/release.yml
Comment (pschicho) [inline .github/workflows/release.yml:1]: I think the tests fail because `WallGoMatrix.m` is now located in `src/Kernel/`. Fixed this.

File: buildPaclet.m
Comment (pschicho) [inline buildPaclet.m:1]: Could this also be run with the workflows. I found this one (https://github.com/WolframResearch/WL-FunctionCompile-CI-Template) but it requires Wolfram ID stored as secrets on the GitHub organisation. And I am a bit insecure if sharing those with the organisation is within bounds of the license.

**Review by pschicho — COMMENTED** [review]
The buildPaclet works well. So I build it locally and add it to the release? Ideally `buildPaclet.m` is run via GitHub. We can look at this in a later PR.

**Review by pschicho — APPROVED** [review]

File: buildPaclet.m
Comment (og113) [reply buildPaclet.m:1]: Oh, this looks interesting. Would be a great solution, if we can get it to work.

**Comment by og113** [comment]
To avoid having to set the version number in multiple places, I have remove it, and the date from WallGoMatrix.m, which means it is no longer printed when the package is loaded. However, Mathematica has the information as a paclet. You can see this by running
```
PacletFind["WallGoMatrix"]
```
after installation. Alternatively, if you want to get the version number explicitly, run
```
Information[PacletFind["WallGoMatrix"], "Version"]
```
Mathematica can deal with multiple versions of the same Paclet, and will by default load the latest one. I think this kind of package management is better than what we had before with just moving folders into Applications, especially with regard to changing versions of the code.

File: .github/workflows/release.yml
Comment (og113) [reply .github/workflows/release.yml:1]: Thanks, but I have now rejigged things further in the release.yml. See comments below.

**Comment by og113** [comment]
Note, I also had to remove the tag from the filenames to get the URL to be always the same one:
https://github.com/Wall-Go/WallGoMatrix/releases/latest/download/WallGoMatrix.paclet
I did the same for the zip file, which I don't know if we want to keep, but having a fixed filename again allows us to use a fixed URL as the location to download the latest version:
https://github.com/Wall-Go/WallGoMatrix/releases/latest/download/WallGoMatrix.zip

**Review by pschicho — APPROVED** [review]

**Pull request merged**